source/sorted_table_model.ts

```diff
@@ -65,30 +66,82 @@ export class SortedTableModel extends TableModel {
   public constructor(source: TableModel, comparator?: Comparator,
       columnOrder?: ColumnOrder[]) {
     super();
+    this.translatedTable = new TranslatedTableModel(source);
+    if(comparator) {
+      this.comparator = comparator;
+    } else {
+      this.comparator = new Comparator();
+    }
+    if(columnOrder) {
+      this.order = columnOrder.slice();
+    } else {
+      this.order = [];
+    }
+    this.sort(source);
   }
 
   /** Returns the column sort order. */
   public get columnOrder(): ColumnOrder[] {
-    return null;
+    return this.order.slice();
   }
 
   /** Sets the order that the columns are sorted by. */
-  public set columnOrder(columnOrder: ColumnOrder[]) {}
+  public set columnOrder(columnOrder: ColumnOrder[]) {
+    this.order = columnOrder.slice();
+  }
 
   public get rowCount(): number {
-    return 0;
+    return this.translatedTable.rowCount;
   }
 
   public get columnCount(): number {
-    return 0;
+    return this.translatedTable.columnCount;
   }
 
   public get(row: number, column: number): any {
-    return null;
+    return this.translatedTable.get(row, column);
   }
 
   public connect(slot: (operations: Operation[]) => void):
       Kola.Listener<Operation[]> {
-    return null;
+    this.translatedTable.connect(slot);
+    return this.dispatcher.listen(slot);
+  }
+
+  private sort(tableModel: TableModel) {
+    const rowOrdering = [];
+    for(let i = 0; i < tableModel.rowCount; ++i) {
+      rowOrdering.push(i);
+    }
+    rowOrdering.sort((a, b) => this.compareRows(a, b));
+    for(let i = 0; i < rowOrdering.length - 1; ++i) {
+      this.translatedTable.moveRow(rowOrdering[i], i);
+      for(let j = i + 1; j < rowOrdering.length; ++j) {
+        if(rowOrdering[j] <= rowOrdering[i]) {
```

Comment:
Is it ```<``` or ```<=```?

Reply:
It's less than because the only value that could have been equal will always be removed by that point.